internal/librarian/generate.go

```diff
@@ -141,13 +141,18 @@ func (r *generateRunner) run(ctx context.Context) error {
 		}
 		prBody += fmt.Sprintf("feat: generated %s\n", libraryID)
 	} else {
+		failedGenerations := 0
 		for _, library := range r.state.Libraries {
 			if err := r.generateSingleLibrary(ctx, library.ID, outputDir); err != nil {
 				// TODO(https://github.com/googleapis/librarian/issues/983): record failure and report in PR body when applicable
 				slog.Error("failed to generate library", "id", library.ID, "err", err)
 				prBody += fmt.Sprintf("%s failed to generate\n", library.ID)
+				failedGenerations++
 			}
 		}
+		if failedGenerations > 0 && failedGenerations == len(r.state.Libraries) {
```

Comment:
nit: it might be more maintainable to keep track of # of successfully generated libraries > 0. We might have cases where we skip a library generation and don't rely on it actually failing. We could invert the check to only open the PR if at least 1 library successfully generates.

Reply:
Opened #1885 to follow up on this